nautobot_floor_plan/template_content.py

```diff
@@ -43,18 +43,38 @@ def buttons(self):
 </a>"""
 
     def detail_tabs(self):
-        """Add a "Floor Plan" tab to the Location detail view if the Location has a FloorPlan associated to it."""
+        """Add a "Floor Plan" tab to the Location detail view or a "Child Floor Plan" tab if the Location has Children with FloorPlans associated to them."""
         location = self.context["object"]
-        try:
-            location.floor_plan  # pylint: disable=pointless-statement
-            return [
+
+        # Initialize the tabs list
+        tabs = []
+
+        # Determine conditions
+        has_floor_plan = getattr(location, "floor_plan", None) is not None
+        has_children = location.children.exists()
+        has_child_floor_plans = location.children.filter(floor_plan__isnull=False).exists()
+
+        # Add "Floor Plan" tab if applicable
+        if has_floor_plan:
+            tabs.append(
                 {
                     "title": "Floor Plan",
                     "url": reverse("plugins:nautobot_floor_plan:location_floor_plan_tab", kwargs={"pk": location.pk}),
                 }
-            ]
-        except ObjectDoesNotExist:
-            return []
+            )
+
+        # Add "Child Floor Plan(s)" tab if applicable
+        if has_children and has_child_floor_plans:
```

Comment:
This still seems redundant. Doesn't the fact that there are child floor plans imply the location has children?

Reply:
sure does, thanks!